hardware/arduino/cores/arduino/HardwareSerial.h

```diff
@@ -50,11 +60,16 @@ class HardwareSerial : public Stream
       uint8_t rxen, uint8_t txen, uint8_t rxcie, uint8_t udre, uint8_t u2x);
     void begin(long);
     void end();
-    virtual int available(void);
-    virtual int peek(void);
-    virtual int read(void);
-    virtual void flush(void);
-    virtual void write(uint8_t);
+    
+    // @todo I think these only need to be declared virtual if HWSerial's going to be extended
```

Comment:
No, virtual is not needed.

If a derived object implements `peek()` this is the one the virtual methods will call. For that object to use the parent versions, it'll have to explicitly call `derived::peek()`.

I wouldn't use `virtual` for another reason: A client may want to use the derived version without implementing their own `using derived::peek;`.